storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -2132,32 +2144,29 @@ fseg_alloc_free_page_low(
 		/* fsp_alloc_free_page() invoked fsp_init_file_page()
 		already. */
 		return(block);
-		/*-----------------------------------------------------------*/
-	} else {
-		/* 7. We allocate a new extent and take its first page
-		======================================================*/
-		ret_descr = fseg_alloc_free_extent(seg_inode, iblock, &xdes,
-						   space, mtr, err);
-
-		if (!ret_descr) {
-			ut_ad(!has_done_reservation || *err);
-			return nullptr;
-		} else {
-			ret_page = xdes_get_offset(ret_descr);
-		}
 	}
 
+	/* 4. In worst case, try to allocate a new extent and
+	take its first page */
+	ret_descr = fseg_alloc_free_extent(seg_inode, iblock, &xdes,
+					   space, mtr, err);
```

Comment:
If we added `} else {` for this one, we could remove `goto alloc_done` from the end of `if (reserved - used > 0) {`.